src/rest.cpp

```diff
@@ -107,6 +107,27 @@ static CTxMemPool* GetMemPool(const std::any& context, HTTPRequest* req)
     return node_context->mempool.get();
 }
 
+/**
+ * Get the node context chainstatemanager.
+ *
+ * @param[in]  req The HTTP request, whose status code will be set if node
+ *                 context chainstatemanager is not found.
+ * @returns        Pointer to the chainstatemanager or nullptr if none found.
+ */
+static ChainstateManager* GetChainman(const std::any& context, HTTPRequest* req)
```

Comment:
Would be happy to add to PR if you supply a simple enough diff!

Reply:
Something like this maybe, if have to retouch: https://github.com/fjahr/bitcoin/commit/69187f82b1227cc33bc5733c742b8f8b9fcefea3

Reply:
This creates a copy of chainman, no?

Reply:
@fjahr I think the branch is invalid because you're trying to move from the global `g_chainman`.

In general, the usage of algebraic sum types like std::optional or std::variant isn't for prettiness - it's to entirely remove the possibility of being in an invalid program state.

Reply:
Ok, never mind, I guess since there is no optional reference in std this wasn't a good idea after all.